src/framework/caspool.cpp

```diff
@@ -77,6 +79,38 @@ namespace uima {
   }
 
 
+  CASPool::CASPool(AnnotatorContext *anContext, const AnalysisEngineDescription &taeSpec,
+                   size_t numInstances) : iv_vecAllInstances(),
+                                          iv_vecFreeInstances(),
+                                          iv_numInstances(numInstances),
+                                          iv_pCasDef(nullptr),
+                                          iv_pOwner(anContext) {
+    iv_pCasDef = internal::CASDefinition::createCASDefinition(taeSpec);
+    if (iv_pCasDef == nullptr) {
+      UIMA_EXC_THROW_NEW(CASPoolException,
+                         UIMA_ERR_CASPOOL_CREATE_CASDEFINITION,
+                         UIMA_MSG_ID_EXC_CREATE_CASPOOL,
+                         UIMA_MSG_ID_EXC_CREATE_CASPOOL,
+                         ErrorInfo::unrecoverable);
+    }
+
+    for (size_t i = 0; i < numInstances; i++) {
+      CAS *pCas = uima::internal::CASImpl::createCASImpl(*iv_pCasDef, false);
+      if (pCas == nullptr) {
+        UIMA_EXC_THROW_NEW(CASPoolException,
+                           UIMA_ERR_CASPOOL_CREATE_CAS,
+                           UIMA_MSG_ID_EXC_CREATE_CASPOOL,
+                           UIMA_MSG_ID_EXC_CREATE_CASPOOL,
+                           ErrorInfo::unrecoverable);
+      }
+      CAS *initialView = pCas->getInitialView();
```

Comment:
this seems wrong. Why only the owner of the initialview? Shouldn't be the owner of the whole CAS? What about the other views?

Reply:
I believe that's the way the framework constructs the CAS Pool. If you look at the old constructor, it's not adding the new base CAS into the pool, but only its initial view.

Reply:
I misunderstood your comment, but after checking with the Java version I see you're correct. We should be setting the owner for the base CAS instead. I am fixing that.